scipy/optimize/_linprog_util.py

```diff
@@ -1126,13 +1126,20 @@ def vstack(blocks):
             A_eq[:, i_nolb] *= -1
 
     # upper bound: add inequality constraint
-    i_newub = np.nonzero(ub_some)[0]
+    i_newub, = ub_some.nonzero()
     ub_newub = ubs[ub_some]
```

Comment:
Just curious - why did the part below need to be changed? 
This is not currently benchmarked - do you think it should be?

Reply:
Your update is exactly right. The rationale of keeping the dense case as-is was that in-place updates are cheap for dense arrays.

More benchmark coverage is always welcome, though I don't have a sense for how performance-sensitive this code path is.